### PR TITLE
#186374352 : Support multipart outgoing

### PR DESCRIPTION
**Pull request by xinghengwang**

### CODE REVIEW

File: lib/moesif_rack/moesif_helpers.rb [hunk 1]
```diff
@@ -38,4 +38,23 @@ def format_replacement_body(replacement_body, original_body)
...
+    tempfile = Rack::Multipart::Parser::TEMPFILE_FACTORY
+    bufsize = Rack::Multipart::Parser::BUFSIZE
+    query_parser = Rack::Utils.default_query_parser
+    result = Rack::Multipart::Parser.parse(io, sanitized_multipart_form_data.length, content_type, tempfile, bufsize,
+                                           query_parser)
+
+    log_debug('multipart parse result')
+    log_debug(result.inspect)
+
+    # this is a hash shold be treated as JSON down the road.
```
Comment (praves77) [inline lib/moesif_rack/moesif_helpers.rb]: typo here `shold` -> `should`

**Review by praves77 — APPROVED** [review]
lftm

File: moesif_capture_outgoing/httplog/http_log.rb [hunk 2]
```diff
@@ -44,18 +49,30 @@ def transform_response_code(response_code_name)
...
         req_body_transfer_encoding = nil
         req_body = nil
 
-        if @log_body_outgoing && (req_body_string && req_body_string.length != 0)
+        if @log_body_outgoing && (not req_content_type.nil?) && (req_content_type.downcase.include? 'multipart/form-data')
+          @moesif_helpers.log_debug 'outgoing request is multipart, parsing req_body_from_stream'
+          begin
+            req_body = @moesif_helpers.parse_multipart(req_body_from_stream, req_content_type)
+          rescue StandardError
+            @moesif_helpers.log_debug 'outgoing request is multiple, but failed to process req_body_from_stream: ' + req_body_from_stream.to_s
```
Comment (keyur9) [inline moesif_capture_outgoing/httplog/http_log.rb]: should be `outgoing request is multipart`

**Review by keyur9 — APPROVED** [review]
lgtm

**Pull request merged**